src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -359,35 +359,101 @@ public static Vector128<T> ConditionalSelect<T>(Vector128<T> condition, Vector12
         /// <param name="vector">The vector to convert.</param>
         /// <returns>The converted vector.</returns>
         [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<double> ConvertToDouble(Vector128<long> vector)
         {
-            Unsafe.SkipInit(out Vector128<double> result);
+            if (Sse2.IsSupported)
+            {
+                // Based on __m256d int64_to_double_fast_precise(const __m256i v)
+                // from https://stackoverflow.com/a/41223013/12860347. CC BY-SA 4.0
```

Comment:
The three bits of code here are new algorithms and are significantly faster than the previous.

They are also correct where-as various inputs on the old algorithm would actually return a different result as compared to the scalar versions.